tmt/steps/provision/__init__.py

```diff
@@ -923,6 +923,16 @@ def setup(self) -> None:
 
         Setup the guest after it has been started. It is called after :py:meth:`Guest.start`.
         """
+        from tmt.steps.execute.internal import effective_pidfile_root
```

Comment:
One thing I noticed: there's `Guest.setup()` and `GuestSsh.setup()`. The former is pretty empty, and you are adding first code into it; the latter already contains some code.

Byw `GuestSsh.setup()` does not call its base class implementation, it lacks `super().setup()` call. Looks like an omission which was fine, `Guest.setup()` did nothing so it didn't hurt. Now it does, and we need to call it. Please, add the `super()` call to `GuestSsh` so we have the chain complete.

Reply:
So with this you mean it'd better if I move the code from `Guest.setup()` to the `GuestSsh` one and add the super().setup() ? or only do the last part in the Guest.setup() function?

Reply:
Nono, I think `Guest` is the right place. We just need one extra call at the beginning of `GuestSsh.setup`, and that would be calling its base class implementation of the method it overrides.

Reply:
Got it, code updated, thanks!